docs/walkthrough.md

```diff
@@ -0,0 +1,204 @@
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+-->
+
+# From Model to Deployment
+
+Given a trained model, how do I take it and deploy it at-scale with an
+optimal configuration using Triton Inference Server? This document is
+here to help answer that.
+
+There are a few steps in this process, and there is no single answer for
+everything, but here is the common flow:
+
+1. Is my model compatible with Triton?
+    - If your model falls under one of Triton's [supported backends](https://github.com/triton-inference-server/backend), then we can simply try to deploy the model as described in the [Quickstart](https://github.com/triton-inference-server/server/blob/main/docs/quickstart.md) guide. For the ONNXRuntime, TensorFlow SavedModel, and TensorRT backends, the model configuration can be completely inferred from the model using Triton's [AutoComplete](https://github.com/triton-inference-server/server/blob/main/docs/model_configuration.md#auto-generated-model-configuration) feature, so no `config.pbtxt` file is required. For other backends, refer to the [Minimal Model Configuration](https://github.com/triton-inference-server/server/blob/main/docs/model_configuration.md#minimal-model-configuration) required to get started.
+    - If your model does not come directly from a supported backend, but is executable through Python; you may be interested in writing a simple python script for running your model through the [Python Backend](https://github.com/triton-inference-server/python_backend).
+    - Otherwise, you can create a [Custom Backend](https://github.com/triton-inference-server/backend/blob/main/examples/README.md) for executing your model.
+
+2. Does my model perform well?
+    - Assuming you were able to deploy your model on Triton, the next step is to benchmark the performance of your model. Triton's [Perf Analyzer](https://github.com/triton-inference-server/server/blob/main/docs/perf_analyzer.md) tool specifically fits this purpose. Here is a simplified output for demonstration purposes:
+
+    ```
+    # NOTE: "my_model" represents the model currently being served by 
+    #       Triton in the background from the previous step.
+
+    $ perf_analyzer -m my_model
+
+    ...
+
+    Inferences/Second vs. Client Average Batch Latency
+    Concurrency: 1, throughput: 482.8 infer/sec, latency 12613 usec
+    ```
+
+    - The defintion of "performing well" is subject to change for each use case. There are many variables that can be tweaked just within your Triton configuration (`config.pbtxt`) to obtain different results.
```

Comment:
Provide an example? Two common use cases is to get their throughput (number of inference requests completed per second) to a certain target or reduce their latency.